### DISCUSSION TITLE
(Submod) - April First

### DISCUSSION THREAD

**Discussion by wboorman** [Monika-After-Story/MonikaModDev#7822] [general - submod / external content]
Monika After Story: April First ([link](https://github.com/DrakeTheDuelist/MAS-AprilFirst))

Description: Monika has a surprise for the player on April 1st...

As the name would suggest, there is a bit of a time crunch on getting this looked at and perfected. Plus, I haven't ever deployed a submod via github before. (_Dreams and Nightmares_ is suffering from a bit of feature creep in service of future proofing it. Sue me.) I'd like for the early adopters to have at least a day or so wondering what the mod does before April 1st.

Also, as it's very obviously an April Fool's joke, the source code will COMPLETELY spoil the main joke, as might the fact that there's an MP3 file in there. MP3 file... April 1st... yeah, doesn't take a rocket scientist to see where this is going. And no, it's not a jump scare. In fact, depending on your perspective, it might be worse...

Still, I think I make a decent case that this isn't just some lazy meme cash-in, but something that's actually in character for Monika to do. I strive to keep Monika in character to the best of my knowledge, even if some thoughts I have about Monika's personality aren't universally recognized as canon. (See also _Dreams and Nightmares_.) Perhaps I go a bit overboard in some areas. I am _extremely_ well-versed in taking any character I write into dark places, so my rendition of Monika tends to cry at the drop of a hat. And I won't lie, even _I_ kinda' felt some of those moments during writing.

**Comment by MadFoot8** [comment]
[code]
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/script-ch30.rpy", line 1475, in script call
  File "game/event-handler.rpy", line 2516, in script call
  File "game/Submods/AprilFirst/NGGYU.rpy", line 49, in script call
    call rickroll_song_nggyu(do_prank=True)
  File "game/Submods/AprilFirst/NGGYU.rpy", line 87, in script call
    call rickroll_song_nggyu_analysis(from_prank=True)
  File "game/Submods/AprilFirst/NGGYU.rpy", line 111, in script call
    call rickroll_monika_interpretation
  File "game/Submods/AprilFirst/NGGYU.rpy", line 293, in script
    $ nickname = ", " + nickname
  File "game/Submods/AprilFirst/NGGYU.rpy", line 293, in <module>
    $ nickname = ", " + nickname
TypeError: coercing to Unicode: need string or buffer, function found

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "game/script-ch30.rpy", line 1475, in script call
  File "game/event-handler.rpy", line 2516, in script call
  File "game/Submods/AprilFirst/NGGYU.rpy", line 49, in script call
    call rickroll_song_nggyu(do_prank=True)
  File "game/Submods/AprilFirst/NGGYU.rpy", line 87, in script call
    call rickroll_song_nggyu_analysis(from_prank=True)
  File "game/Submods/AprilFirst/NGGYU.rpy", line 111, in script call
    call rickroll_monika_interpretation
  File "game/Submods/AprilFirst/NGGYU.rpy", line 293, in script
    $ nickname = ", " + nickname
  File "C:\Users\Saran\Desktop\DDLC MaS\renpy\ast.py", line 814, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "C:\Users\Saran\Desktop\DDLC MaS\renpy\python.py", line 1719, in py_exec_bytecode
    exec bytecode in globals, locals
  File "game/Submods/AprilFirst/NGGYU.rpy", line 293, in <module>
    $ nickname = ", " + nickname
TypeError: coercing to Unicode: need string or buffer, function found

Windows-8-6.2.9200
Ren'Py 6.99.12.4.2187
Monika After Story 0.12.1
[/code]

**Comment by NegativeOne404** [comment]
Yooooo hahahahahaha, i loved this one, I think it really suits with her personality as you said. I really liked! Ah btw, there was an error when she tried to get our name, fortunately ignoring this one is safe. Here is the traceback:
[traceback.txt](https://github.com/Monika-After-Story/MonikaModDev/files/6242899/traceback.txt)

**Comment by CrocandCo** [comment]
I did get an error too, it does seem to occur on the last line for getting the nickname of the player. So those who haven't installed this sub-mod may get the chance to get a glitchless version in the next year?

**Comment by Abc2557** [comment]
I loved that! Definitely one of the better april fools pranks I've seen, haha ^^
